config.go

```diff
@@ -0,0 +1,132 @@
+	// Default value is TopicPolicyCreate.
+	TopicPolicy TopicPolicy
+	// If true then KafkaLock won't alter Sarama's timeouts configuration to preferred/recommended values.
+	// Be aware that the recommended values require you to adjust Kafka's configuration (group.min.session.timeout.ms).
+	NoTimeoutsTweaking bool
+	// If true then KafkaLock won't alter your Kafka's ClientID at all. You are advised to provide your own anyway.
+	NoClientIdSuffix bool
+	// String identifier that is unique for all KafkaLocker instances. It mustn't change between application restarts.
+	// Currently it is used only for default FlapGuard, so if you provide your own implementation
+	// then you may leave it empty.
+	AppId string
+	// Utility class that protects the KafkaLocker against application "flapping". Leave nil to use the
+	// default implementation. Check the FlapGuard interface for more details.
+	FlapGuard FlapGuard
+	// You can set this to customize logging behaviour of KafkaLock. Default is to log nothing (NopLogger).
+	Logger *Logger
+}
+
+func processConfig(c *Config) (*Config, error) {
+	if c.MaxHolders < 1 {
+		return nil, errors.New("Config.MaxHolders must be greater then or equal to 1")
+	}
+
+	if c.ReplicationFactor < 1 {
+		return nil, errors.New("Config.ReplicationFactor must be greater then or equal to 1")
+	}
+
+	if c.Logger == nil {
+		c.Logger = NewNopLogger()
+	}
+
+	if c.KafkaConfig == nil {
+		c.KafkaConfig = sarama.NewConfig()
+	}
+
+	kafkaConfig := *c.KafkaConfig
+	c.KafkaConfig = &kafkaConfig
+
+	kafkaConfig.Metadata.Full = false
+	kafkaConfig.Metadata.RefreshFrequency = 0
+	kafkaConfig.Consumer.Return.Errors = false
+	kafkaConfig.Consumer.Group.Rebalance.Strategy = &kafkaLockBalanceStrategy{c.Logger}
+
+	if !c.NoTimeoutsTweaking {
+		kafkaConfig.Consumer.Group.Session.Timeout = time.Second * 3
+		kafkaConfig.Consumer.Group.Heartbeat.Interval = time.Second
+		kafkaConfig.Consumer.Group.Rebalance.Timeout = time.Second
+		kafkaConfig.Consumer.Retry.Backoff = time.Millisecond * 500
+	}
+
+	if !c.NoClientIdSuffix {
+		if kafkaConfig.ClientID == "" {
+			kafkaConfig.ClientID = "kafkalock"
+		} else {
+			kafkaConfig.ClientID = c.KafkaConfig.ClientID + ".kafkalock"
+		}
+	}
+
+	if c.FlapGuard == nil {
+		if c.AppId == "" {
+			return nil, errors.New("unique static AppId must be set to use the default FlapGuard")
+		}
+
+		dir := path.Join(os.TempDir(), "kafkalock")
+		err := os.MkdirAll(dir, 0775)
+		if err != nil {
+			return nil, err
+		}
+
+		c.FlapGuard, err = NewFileFlapGuard(c.Logger, path.Join(dir, c.AppId), 3, true)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return c, nil
+}
+
+// Returns a new Config struct filled with the default values.
+func NewConfig(bootstrapServers []string, topic string) *Config {
+	return &Config{
+		BootstrapServers:  bootstrapServers,
+		Topic:             topic,
+		MaxHolders:        1,
+		ReplicationFactor: 3,
```

Comment:
Shouldn't this be configurable?

Reply:
It is ... why would you think that it's not? This is just a default constructor - you can change any value of the Config struct as you like.